enhancements/proposals/1405-alternator/README.md

```diff
@@ -0,0 +1,182 @@
+
+### Goals
+
+- Expose Alternator service using a load balanced endpoint (Service)
+- Expose Alternator over secure and static port
+- Expose Alternator on insecure port only if explicitly requested
+- Ensure e2e coverage for Alternator
+- Expose Alternator together with CQL (not exclusively)
+- Deprecate `scyllaclusters.scylla.scylladb.com/v1.spec.alternator.port`
+- Enable transition period and be backwards compatible with existing clusters
+
+### Non-Goals
+
+- Exposing Alternator on every node individually and relying on client side load balancing
+- Supporting host networking tweaks
+- Disabling CQL when Alternator is enabled
+- Enabling Alternator on insecure port only
+
+## Proposal
+
+I propose to expose Alternator using the existing identity Service (`<sc-name>-client`) on port `8043` (https).
+Alternator API won't be exposed by default, only on an explicit request.
+When Alternator API is enabled, the insecure port `8000` will be disabled by default, unless explicitly opted-in.
+Disabling the insecure port won't have effect on the old setup with manually specified port.
+Any Alternator settings won't have effect on how / if CQL is exposed.
+This is primarily meant to make sure
+the Scylla Operator or the administrator can still execute maintenance task using CQL, like in
+[#1358](https://github.com/scylladb/scylla-operator/issues/1358).
+Also, the Alternator [authorization](https://enterprise.docs.scylladb.com/stable/alternator/compatibility.html#authorization)
+is managed through CQL by updating `system_auth.roles` table.
+
+In the same release `scyllaclusters.scylla.scylladb.com/v1.spec.alternator.port` will be deprecated, so
+we may ignore the value in a future release. In this release, and with the new changes, it is intended for it to retain the existing behaviour for backward compatibility.
+
+### User Stories
+
+#### Automatic certificate
+As a user I want to enable Alternator API, and have serving certificate and key automatically generated.
+To configure the clients to trust the certificates I'll use the provided serving CA from a ConfigMap.
+
+#### Custom certificate
+As a user I want to enable Alternator API and provide my own serving certificate and either ensure it is signed 
+by a CA that clients implicitly trust or provide them with an appropriate CA on my own.
+
+#### Ingress
+As a user I want to expose Alternator API using an Ingress.
+This allows me to expose it externally without allocating a dedicated IP.
+
+#### Load Balancer Service
+As a user I want to expose Alternator API on Service with a load balancer.
+This allows me to direct traffic directly through a platform LB on a dedicated IP.
+
+### Risks and Mitigations
+
+Not known.
+
+## Design Details
+
+### API changes for ScyllaClusters.scylla.scylladb.com/v1
+
+The following API changes show how we'd approximately extend `scyllaclusters.scylla.scylladb.com/v1.spec.alternator` specification. Only new fields are shown for existing structs.
```

Comment:
I'm not sure I'm seeing the port definition (8043 by default, I assume) ?

Reply:
yep we chose `8043` - described in https://github.com/scylladb/scylla-operator/pull/1406/files#diff-e7c253eebc3e6c9a4c7923ddf75c5c939cfe058d9cf01f085803b4974c2f378eR38